### ISSUE TITLE
Watcher not properly aware of file changes - different behavior before and after daemon restart

### ISSUE THREAD

**Issue by xHN35RQ** [syncthing/syncthing-gtk#102]
I was unable to get filesystem monitoring to work, and so I used "syncthing-gtk -v" to see what's going on. I noticed that:

1. watcher can see folder changes and reports them
2. watched folders will not sync with any connected devices
3. watcher cannot see file changes
4. Thus, no file changes are synced to connected devices

This is tested with rescan=0 and also rescan=999999 and neither affected the above outcomes.

I'm on Linux Mint 17.1 (14.04) and inotify seems to be running fine on this system, as I have other apps using inotify successfully.

Are there additional troubleshooting steps I can take?

EDIT: I used "syncthing-gtk -d" and noticed the following:

5) immediately upon activating file-system monitoring (but BEFORE clicking "restart" button to save configuration) the following occured:
  * watcher is immediately aware of all file and folder changes 
  * all file system changes are immediately synced to all connected devices  

6) after restart, watcher was still aware of all file and folder changes but none of the changes would sync to connected devices
7) watcher seems to report that all connected devices are being updated after a filesystem change but nothing changes on the connected device

I was able to duplicate the behavior of 5 & 6 above with another folder, so I think that synthing-gtk is not saving the configuration correctly upon restart.

**Comment by kozec** [comment]
Hmmm... Watcher shouldn't report updates on remote devices, there is no way how to know about that...

Anyway, if I understood correctly, after syncthing daemon is restarted, you can see watcher reporting  changes in debug log, but files are not transfered to remote device? 
If run `syncthing-gtk -d` in terminal and do `touch /tmp/syncrhonized_folder/new_file`  I can see following:
```
tom@silver:~$ syncthing-gtk -d 2>&1 | grep Watcher
V Watcher       Watching /tmp/syncrhonized_folder
D Watcher       File Changed test /tmp/syncrhonized_folder/new_file > new_file
```
and after second or so number of files reported on remote machine increases and new_file appears.
What part is different on your side?

**Comment by xHN35RQ** [comment]
> Hmmm... Watcher shouldn't report updates on remote devices, there is no way how to know about that...

Sorry I meant syncthing-gtk -d is showing updates on remote devices even though nothing is transferred.

> Anyway, if I understood correctly, after syncthing daemon is restarted, you can see watcher reporting changes in debug log, but files are not transfered to remote device?

Yes, that is correct. If I add a folder and set it to disable the scan, and enable the file system monitoring, and click save, immediately I am asked to Click the restart button. However before I do, I can see Watcher is correctly noting changed files and updating live to remote devices. If I postpone clicking the restart button, Watcher and Syncthing-GTK will perform normally as expected. Once I click the restart button and force the restart is when I begin to have trouble.

> If run syncthing-gtk -d in terminal and do touch /tmp/syncrhonized_folder/new_file I can see following... What part is different on your side?

If I do the same as you I can see the same thing, Watcher notes the appropriate changes and files arrive properly... it performs the same *UNTIL* the restart button is clicked... then after that the part that is different is:

1) file and Folder counts do not increase in the local Syncthing-GTK window
2) counts do not increase on remote devices
2) Files and folders never arrive on remote devices

**Comment by kozec** [comment]
No, this shouldn't have anything to do with configuration. When daemon is restarted, so is watcher - it will unregister all inotify handlers and add them again. Probably this is where it goes wrong.

Can you see ´D Watcher       File Changed´ messages after restart?

**Comment by xHN35RQ** [comment]
Yes I can see `D Watcher File Changed` messages after daemon is restarted. The file counts do not update within the local Syncthing-GTK window, neither do the files update on remote devices. The condition remains the same after restarting ST-GTK. Thanks!

**Comment by kozec** [comment]
Just to be sure, what version of Syncthing (daemon) are you running?

**Comment by xHN35RQ** [comment]
I was on v0.10.24 and then updated to v0.10.25 - both versions display this behaviour.

**Comment by kozec** [comment]
Should be fixed by 6e7c129800427f5e6d0fb95e38c40bccabb5fe5f and 8605a4131e8e8b309463f78c572eddb9809918a8.
